epochStart/metachain/systemSCs.go

```diff
@@ -178,14 +178,15 @@ func (s *systemSCProcessor) ProcessSystemSmartContract(
 	nonce uint64,
 	epoch uint32,
 ) error {
-	if s.enableEpochsHandler.IsSwitchHysteresisForMinNodesFlagEnabledForCurrentEpoch() {
+	currentEpoch := s.enableEpochsHandler.GetCurrentEpoch()
+	if s.enableEpochsHandler.IsSwitchHysteresisForMinNodesFlagEnabledInSpecificEpochOnly(currentEpoch) {
```

Comment:
from what i see we use current epoch in all calls, should we think of getting current epoch inside enableEpochsHandler component as default behavior or something?

Reply:
or there are other flags that will use a different epoch? and will be updated in a next PR

Reply:
that was the main change for the enable epochs handler. The previous implementation did this internally. :D 
The refactoring is needed so we can ask the component if a particular flag is active or not depending on a provided epoch. Real case usage: we have a component that will need to answer the status on some flags for older epochs.